### PR TITLE
Implement hash_filegroup in the build language

### PR DESCRIPTION
**Pull request by peterebden**
Gets rid of a lot of internal specialisation nonsense and makes it work for remote execution.

### CODE REVIEW

File: rules/misc_rules.build_defs [hunk 1]
```diff
@@ -286,16 +286,16 @@ def hash_filegroup(name:str, srcs:list=None, deps:list=None, exported_deps:list=
     return build_rule(
         name=name,
         srcs=srcs,
-        cmd='',
+        output_dirs = ['_out'],
+        cmd = 'mkdir _out; for S in $SRCS; do B="$(basename $S)"; mv "$S" "_out/${B%.*}-$(shasum -a 256 $S | cut -d " " -f 1 ).${B##*.}"; done',
```
Comment (Tatskaari) [inline rules/misc_rules.build_defs:290]: I have no idea what this does but I trust you that it works.

File: test/filegroup/BUILD [hunk 2]
```diff
@@ -12,3 +12,14 @@ plz_e2e_test(
     name = "filegroup_concurrent_build_test",
     cmd = "plz clean //test/filegroup:gen && plz build //test/filegroup/many:all",
 )
+
+hash_filegroup(
+    name = "hashed",
+    srcs = ["hash_filegroup_test.txt"],
```
Comment (Tatskaari) [inline test/filegroup/BUILD:18]: Might be nice to have 2 files here so we trigger that loop in the bash

File: rules/misc_rules.build_defs (hunk 1 above)
Comment (peterebden) [reply rules/misc_rules.build_defs:290]: It turns a filename like `script.js` into `script-a13ffe6ee73459f04e180132a3e38c3e5fc145e27b4a5a4613960b01058680aa.js`

File: test/filegroup/BUILD (hunk 2 above)
Comment (peterebden) [reply test/filegroup/BUILD:18]: Yeah fair, don't quite have time now though, will revisit later.

**Pull request merged**